module/zfs/zfs_ctldir.c

```diff
@@ -1064,7 +1054,8 @@ zfsctl_snapshot_mount(struct path *path, int flags)
 	zfs_sb_t *snap_zsb;
 	zfs_snapentry_t *se;
 	char *full_name, *full_path;
-	char *argv[] = { "/bin/sh", "-c", NULL, NULL };
+	char *argv[] = { "/usr/libexec/zfs/z_usermodehelper", "mount",
```

Comment:
I thought `/usr/libexec` was just as much a standard location today, but apparently it's only recently been added to FHS.

Yes, calling `/sbin/mount.zfs` and `/sbin/umount.zfs` directly should be doable.